blockchain/validate.go

```diff
@@ -1369,127 +1342,108 @@ func checkCoinbaseUniqueHeightWithAddress(blockHeight int64, minReqOutputs, inde
 		}
 	}
 	if len(nullData) > maxUniqueCoinbaseNullDataSize {
-		str := fmt.Sprintf("block %s coinbase output 1 pushes %d bytes which "+
-			"is more than allowed value of %d", block.Hash(), len(nullData),
-			maxUniqueCoinbaseNullDataSize)
+		str := fmt.Sprintf("block %s coinbase output %d pushes %d bytes which "+
+			"is more than allowed value of %d", block.Hash(), nullDataOutIdx,
+			len(nullData), maxUniqueCoinbaseNullDataSize)
 		return ruleError(ErrFirstTxNotCoinbase, str)
 	}
 	if len(nullData) < 4 {
-		str := fmt.Sprintf("block %s coinbase output 1 pushes %d bytes which "+
-			"is too short to encode height", block.Hash(), len(nullData))
+		str := fmt.Sprintf("block %s coinbase output %d pushes %d bytes which "+
+			"is too short to encode height", block.Hash(), nullDataOutIdx,
+			len(nullData))
 		return ruleError(ErrFirstTxNotCoinbase, str)
 	}
 
 	// Check the height and ensure it is correct.
 	cbHeight := binary.LittleEndian.Uint32(nullData[0:4])
 	if cbHeight != uint32(blockHeight) {
 		header := &block.MsgBlock().Header
-		str := fmt.Sprintf("block %s coinbase output 1 encodes height %d "+
+		str := fmt.Sprintf("block %s coinbase output %d encodes height %d "+
 			"instead of expected height %d (prev block: %s, header height %d)",
-			block.Hash(), cbHeight, uint32(blockHeight), header.PrevBlock,
-			header.Height)
+			block.Hash(), nullDataOutIdx, cbHeight, uint32(blockHeight),
+			header.PrevBlock, header.Height)
 		return ruleError(ErrCoinbaseHeight, str)
 	}
 
 	return nil
 }
 
-// checkCoinbaseUniqueHeightWithTreasuryBase checks to ensure that for all
-// blocks height > 1 the treasurybase contains the height encoding to make
-// treasurybase hash collisions impossible. This is the new function that
-// assumes that there is no treasury address.
-func checkCoinbaseUniqueHeightWithTreasuryBase(blockHeight int64, block *dcrutil.Block) error {
+// checkTreasurybaseUniqueHeight ensures that for all blocks height > 1 the
+// treasurybase contains the height encoding to make treasurybase hash
+// collisions impossible.
+func checkTreasurybaseUniqueHeight(blockHeight int64, block *dcrutil.Block) error {
+	// Block 0 and 1 are special and don't need the treasurybase height checks.
+	if blockHeight < 2 {
+		return nil
+	}
+
 	if len(block.MsgBlock().STransactions) == 0 {
-		str := fmt.Sprintf("checkCoinbaseUniqueHeightWithTreasuryBase: " +
-			"has 0 stake transactions")
+		str := fmt.Sprintf("checkCoinbaseUniqueHeightWithTreasuryBase: has 0 " +
```

Comment:
Shouldn't the block hash and height be included in the error message here?

Reply:
Good point.  I had just copied the existing code, but it makes sense to provide more details to make it more obvious what happened should it ever be hit.  Updated.